projectq/ops/_gates.py

```diff
@@ -74,17 +77,29 @@ def matrix(self):
 
 class XGate(SelfInverseGate):
```

Comment:
Done. However I had to define the XGate commutable gates exclusively within a get_method. This is a bit of a messy fix, but the alternative is to look into defining commutation rules once all gates are defined. Which is a project I don’t want to look into right now.

Reply:
If not already done, could you add this remark as a comment somewhere in the code (e.g. maybe at the top of the file), so that we know what a possible next step could be?

Reply:
Done. As will be seen in next commit.